### DISCUSSION TITLE
Swatten Hybrid Inverter

### DISCUSSION THREAD

**Discussion by lgisler72** [wills106/homeassistant-solax-modbus#723] [Q&A]
Hello, 
Is it possible that you make a ''plugin_swatten'' for Solax_modbus if I provide you the modbus protocol?
I am not enough skilled to do this on my own so  it will be wery helpfull.
[Swatten inverter Modbus table.xlsx](https://github.com/wills106/homeassistant-solax-modbus/files/14167396/Swatten.inverter.Modbus.table.xlsx)

Thank you

**Comment by wills106** [comment]
Have you done any testing with a tool like Gmodbus to determine if they are holding or input registers etc?
As it doesn't look like that spreadsheet provides that info.

**Reply by lgisler72** [reply]
Hello,
No I didn't try but I have this doc that I think will be useful...
[key info list.pdf](https://github.com/wills106/homeassistant-solax-modbus/files/14230176/key.info.list.pdf)
[Modbus RTU protocol.pdf](https://github.com/wills106/homeassistant-solax-modbus/files/14230177/Modbus.RTU.protocol.pdf)

**Reply by wills106** [reply]
Is your Inverter Single or 3Phase?

**Reply by lgisler72** [reply]
My inverter is a hybrid inverter 3Phase / Type : SiH-8kW-TH

**Reply by wills106** [reply]
Thanks, I'll expose some 3Phase registers then.
I'm almost ready for an initial test. It will only expose a few entities initially, but it's mostly to confirm communications are working.

**Reply by lgisler72** [reply]
Great!
I will try it as soon as possible (in 5hour) and give you a feedback.

**Reply by wills106** [reply]
I have just released [2024.02.5b5](https://github.com/wills106/homeassistant-solax-modbus/releases/tag/2024.02.5b5)

If it fails could you follow this post and provide the error log https://github.com/wills106/homeassistant-solax-modbus/issues/729#issuecomment-1942071785

The default baud rate is 9600 for Swatten Inverters.
I wouldn't recommend trying to use a polling frequency less than 5s

**Comment by marbeja** [comment]
In Spain we use single fase Up to 12kW but my 2 Growatt SpF 5000 in
parallel can't work with HA, only SPH

**Reply by wills106** [reply]
I have started to add in SPF support #686 
Do you also have a Swatten Inverter? As this discussion is about adding in support for Swatten Inverters.
